### ISSUE TITLE
Install does not complete; leaves WSOD

### ISSUE THREAD

**Issue by Steve4ASU** [freeform/civicrm-drupal-pantheon#1]
Installed your folder of patched civicrm files to a new Drupal 7.50 site on Pantheon. When I run the install program, it gives me green lights to install, but after clicking the install button I get a WSOD. A civicrm.settings.php file is created, but no Civi functions are accessible.

Attached are the errors shown on the Pantheon dashboard.

[Install errors 2_cclidb | Pantheon Dashboard.pdf](https://github.com/freeform/civicrm-drupal-pantheon/files/391664/Install.errors.2_cclidb.Pantheon.Dashboard.pdf)

**Comment by xurizaemon** [comment]
FYI, your db creds are visible in that PDF. Probably not critical as I believe Pantheon keeps each site contained, but ...

**Comment by Steve4ASU** [comment]
Whoops. Thanks for catching that, xurizaemon. I've updated the PDF.

**Comment by herbdool** [comment]
@Steve4ASU I've got a quick fix for you, and I'll update this repo.

after about line 219 in civicrm.settings.php, that is, after:

```
if (!defined('CIVICRM_DOMAIN_ID')) {
  define( 'CIVICRM_DOMAIN_ID', 1);
}
```

Add this closing bracket:

```
} // end Pantheon check
```

I've tested it on a fresh install and that makes it work. Apologies for your hassle.

**Comment by herbdool** [comment]
Fixed in 2be754448abc38cd51c4bb69607fbdd55fd7b1c3. Please test.

**Comment by Steve4ASU** [comment]
Tried your latest changes. No WSOD, but got the following error message during install:

Initialization Error

Array
(
    [callback] => Array
        (
            [0] => CRM_Core_Error
            [1] => simpleHandler
        )

```
[code] => -18
[message] => DB Error: no such table
[mode] => 16
[debug_info] => SELECT *,  config_backend, locales, locale_custom_strings  
```

 FROM civicrm_domain 

 WHERE (  civicrm_domain.id = 1 )  

 [nativecode=1146 *\* Table 'pantheon.civicrm_domain' doesn't exist]
    [type] => DB_Error
    [user_info] => SELECT *,  config_backend, locales, locale_custom_strings  
 FROM civicrm_domain 

 WHERE (  civicrm_domain.id = 1 )  

 [nativecode=1146 *\* Table 'pantheon.civicrm_domain' doesn't exist]
    [to_string] => [db_error: message="DB Error: no such table" code=-18 mode=callback callback=CRM_Core_Error::simpleHandler prefix="" info="SELECT *,  config_backend, locales, locale_custom_strings  
 FROM civicrm_domain 

 WHERE (  civicrm_domain.id = 1 )  

 [nativecode=1146 *\* Table 'pantheon.civicrm_domain' doesn't exist]"]
)

**Comment by herbdool** [comment]
Make sure it's all fresh: remove all civicrm files, including the civicrm.settings.php. And remove the civicrm_\* database tables. Then try again.

If you still get an error please let me know.

**Comment by yosefromano** [comment]
If I am not mistaken, this code requires you to have Redis enabled which is only available with the sandbox and enterprise level and up hosting packages in pantheon. If you are working with a pro level site, it might be an issue.

**Comment by herbdool** [comment]
Right, that's partly true. I've updated it so the default compiled package excludes the patch which will store Smarty templates in Redis cache. If you do want that you can uncomment this line in civicrm_drupal_pantheon.make:

```
;libraries[civicrm][patch][smartyredis] = ./patches/smarty-redis-civi-cache.patch
```

and then run drush make as listed in the README.

The other Redis-related patch won't interfere if you don't have Redis enabled.

**Comment by Steve4ASU** [comment]
First, thank you for working on this. I'm not a coder, so I really appreciate your efforts.

Now, the latest...

Spun up a new Drupal 7 website on Pantheon
Enabled Redis function and confirmed it was running.
Downloaded your latest files.
Uploaded the "civicrm" folder to sites/all/modules (I use an SFTP client; I don't know drush)
Cleared caches, crossed fingers, ran the install script
Hit one snag - the installer didn't see a "sql" folder (which is odd because I could have sworn it existed after the upload, but after a refresh of the file page on the SFTP client saw there wasn't one). So manually created folder called "sql."
Re-check on install page confirmed ready for installation. Clicked install and...

The following error message:

Initialization Error

Array
(
    [callback] => Array
        (
            [0] => CRM_Core_Error
            [1] => simpleHandler
        )

```
[code] => -18
[message] => DB Error: no such table
[mode] => 16
[debug_info] => SELECT *,  config_backend, locales, locale_custom_strings  
```

 FROM civicrm_domain 

 WHERE (  civicrm_domain.id = 1 )  

 [nativecode=1146 *\* Table 'pantheon.civicrm_domain' doesn't exist]
    [type] => DB_Error
    [user_info] => SELECT *,  config_backend, locales, locale_custom_strings  
 FROM civicrm_domain 

 WHERE (  civicrm_domain.id = 1 )  

 [nativecode=1146 *\* Table 'pantheon.civicrm_domain' doesn't exist]
    [to_string] => [db_error: message="DB Error: no such table" code=-18 mode=callback callback=CRM_Core_Error::simpleHandler prefix="" info="SELECT *,  config_backend, locales, locale_custom_strings  
 FROM civicrm_domain 

 WHERE (  civicrm_domain.id = 1 )  

 [nativecode=1146 *\* Table 'pantheon.civicrm_domain' doesn't exist]"]
) 
![screen shot 2016-08-03 at 7 10 37 am](https://cloud.githubusercontent.com/assets/3503698/17368561/80f0e0f0-5949-11e6-8a69-ace7bce4f57a.png)

**Comment by Steve4ASU** [comment]
A thought just occurred. I have been trying to install the 4.6.x branch. I didn't think the 4.7.x branch was ready. Should I be trying the 4.7.x branch instead?

**Comment by Steve4ASU** [comment]
I went ahead and tried an install with the 4.7.x branch. Got this error:

Initialization Error

Array
(
    [callback] => Array
        (
            [0] => CRM_Core_Error
            [1] => simpleHandler
        )

```
[code] => -18
[message] => DB Error: no such table
[mode] => 16
[debug_info] => SELECT id, name, value, domain_id, contact_id, is_domain, component_id, created_date, created_id
```

FROM civicrm_setting
WHERE (domain_id = 1) AND (is_domain = 1)
 [nativecode=1146 *\* Table 'pantheon.civicrm_setting' doesn't exist]
    [type] => DB_Error
    [user_info] => SELECT id, name, value, domain_id, contact_id, is_domain, component_id, created_date, created_id
FROM civicrm_setting
WHERE (domain_id = 1) AND (is_domain = 1)
 [nativecode=1146 *\* Table 'pantheon.civicrm_setting' doesn't exist]
    [to_string] => [db_error: message="DB Error: no such table" code=-18 mode=callback callback=CRM_Core_Error::simpleHandler prefix="" info="SELECT id, name, value, domain_id, contact_id, is_domain, component_id, created_date, created_id
FROM civicrm_setting
WHERE (domain_id = 1) AND (is_domain = 1)
 [nativecode=1146 *\* Table 'pantheon.civicrm_setting' doesn't exist]"]
)

![screen shot 2016-08-03 at 8 31 52 am](https://cloud.githubusercontent.com/assets/3503698/17371516/012885a6-5955-11e6-85cd-09461926ea13.png)

**Comment by herbdool** [comment]
In the last couple days I've tested both the 4.6.x and 4.7.x branch on fresh Pantheon installs and it's worked. So I'm unable to replicate the issue. Sounds like not all the files are getting uploaded with your ftp client so first check that it's not skipping any or failing.

If you browse through the code you can see that the sql folder isn't empty so you will need all those files there.

4.6.x has been better tested than 4.7.x but 4.7 is the latest Civi release.

**Comment by Steve4ASU** [comment]
YES!!! Success!!!

For some reason, the 'sql' folder and files were not being uploaded by the SFTP clients. Neither Cyberduck nor Filezilla apparently did it when uploading the 'civicrm' folder.

What was weird is that after the 'civicrm' folder was uploaded, it looked like the 'sql' folder was there according to the SFTP client's view of the server files. But once I started the install script, the install screen would show a "missing 'sql' folder error. If I then went back to the SFTP client and refreshed the view, the 'sql' folder would indeed be missing.

However, this time I tried to upload just the 'sql' folder from the local to the server, and the folder and files were successfully uploaded. (Thanks, Filezilla, for the side by side comparison view.)

Clicking re-check on the install page gave me the green light to go, so hit install and - voila - received the successful install screen with links to Configuration Checklist and the like.

So there were definitely some files that did not get uploaded in previous attempts.

Thanks for sticking with me.
